src/partially_blindrsa.ts

```diff
@@ -0,0 +1,489 @@
+        const d = inverseMod(e, phi);
+
+        // 7. n = p * q
+        const n = p.mul(q);
+
+        // 7. sk = (n, p, q, phi, d)
+        const sk: BigSecretKey = { n, p, q, d };
+        // 8. pk = (n, e)
+        const pk: BigPublicKey = { e, n };
+        // 9. output (sk, pk)
+        return PartiallyBlindRSA.bigKeyPairToCryptoKeyPair(
+            {
+                secretKey: sk,
+                publicKey: pk,
+            },
+            algorithm,
+            true,
+        );
+    }
+
+    generateKey(
+        algorithm: Pick<RsaHashedKeyGenParams, 'modulusLength' | 'publicExponent'>,
+    ): Promise<CryptoKeyPair> {
+        return PartiallyBlindRSA.generateKey({ ...algorithm, hash: this.params.hash });
+    }
+
+    async verify(
+        publicKey: CryptoKey,
+        signature: Uint8Array,
+        message: Uint8Array,
+        info: Uint8Array,
+    ): Promise<boolean> {
+        const { jwkKey } = await this.extractKeyParams(publicKey, 'public');
+        if (!jwkKey.n) {
+            throw new Error('key has invalid parameters');
+        }
+        const e = sjcl.bn.fromBits(sjcl.codec.base64url.toBits(jwkKey.e));
+        const n = sjcl.bn.fromBits(sjcl.codec.base64url.toBits(jwkKey.n));
+        const pk: BigPublicKey = { e, n };
+
+        // 1. Compute pk_derived = DerivePublicKey(pk, info).
+        const pk_derived = await this.derivePublicKey(pk, info);
+        const pk_derived_key = await crypto.subtle.importKey(
+            'jwk',
+            {
+                ...jwkKey,
+                e: sjcl.codec.base64url.fromBits(pk_derived.e.toBits(0)),
+                n: sjcl.codec.base64url.fromBits(pk_derived.n.toBits(0)),
+            },
+            { name: PartiallyBlindRSA.NAME, hash: this.params.hash },
+            false,
+            ['verify'],
+        );
+
+        // 2. Compute msg_prime = concat("msg", int_to_bytes(len(info), 4), info, msg).
+        const msg_prime = joinAll([
+            new TextEncoder().encode('msg'),
+            int_to_bytes(info.length, 4),
+            info,
+            message,
+        ]);
+
+        // 3. Invoke and output the result of RSASSA-PSS-VERIFY (Section 8.1.2 of [RFC8017]) with (n, e) as pk_derived, M as msg_prime, and S as sig.
+        return crypto.subtle.verify(
+            { name: PartiallyBlindRSA.NAME, saltLength: this.params.saltLength },
+            pk_derived_key,
+            signature,
+            msg_prime,
+        );
+    }
+
+    private async derivePublicKey({ n }: BigPublicKey, info: Uint8Array): Promise<BigPublicKey> {
+        // 1. hkdf_input = concat("key", info, 0x00)
+        // 2. hkdf_salt = int_to_bytes(n, modulus_len)
+        // 3. lambda_len = modulus_len / 2
+        // 4. hkdf_len = lambda_len + 16
+        const hkdf_input = joinAll([new TextEncoder().encode('key'), info, new Uint8Array([0x00])]);
+        const hkdf_salt = i2osp(n, n.bitLength() / 8);
+        const lambda_len = n.bitLength() / 8 / 2;
```

Comment:
protect these two divisions with `Math.ceil`

Reply:
moved to bitshift